### PR TITLE
Updating deployed pods according to the CR spec

### PR DESCRIPTION
**Pull request by dAdAbird**
Now changed only container resources and number of containers in pod

Also made some refactoring

### CODE REVIEW

File: pkg/pxc/service.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Service",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-pxc-proxysql",
+			Namespace: cr.Namespace,
+			Labels: map[string]string{
+				"app":     "pxc",
+				"culster": cr.Name,
```
Comment (delgod) [inline pkg/pxc/service.go]: look like typo

File: pkg/pxc/service.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+			Labels: map[string]string{
+				"app":     "pxc",
+				"cluster": cr.Name,
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port: 3306,
+					Name: "mysql-port",
```
Comment (delgod) [inline pkg/pxc/service.go]: I think we need to use the same port name in both ServicePort definitions(line 63), maybe "mysql" name will be better

File: pkg/pxc/service.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+			Kind:       "Service",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-pxc-nodes",
+			Namespace: cr.Namespace,
+			Annotations: map[string]string{
+				"service.alpha.kubernetes.io/tolerate-unready-endpoints": "true",
+			},
+			Labels: map[string]string{
+				"app":     "pxc",
```
Comment (delgod) [inline pkg/pxc/service.go]: not sure, make a constant here can have sense, up to you

File: pkg/pxc/statefulset.go [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+		Volumes: []corev1.Volume{
+			cfgPV,
+		},
+	}
+
+	pvcObj := corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "datadir",
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
```
Comment (delgod) [inline pkg/pxc/statefulset.go]: hmm, should we specify persistentVolumeReclaimPolicy: Delete 
what do you think?

Comment (dAdAbird) [reply pkg/pxc/statefulset.go]: does it make sense to allow user to define the reclaim policy config with `Delete` defaults? or just make it `Delete`?

Comment (dAdAbird) [reply pkg/pxc/statefulset.go]: Ok, it's not an option of PV, not PVC. And this is inherites from the Storage Class. So there is no way to set it in a Stateful Set. The only way is to get created PVs and redefine this options by updating. But it looks redundant and weird. Way better just use appropriate Storage Class.

File: pkg/pxc/service.go (hunk 3 above)
Comment (dAdAbird) [reply pkg/pxc/service.go]: done

File: pkg/pxc/service.go (hunk 2 above)
Comment (dAdAbird) [reply pkg/pxc/service.go]: done

File: pkg/pxc/service.go (hunk 1 above)
Comment (dAdAbird) [reply pkg/pxc/service.go]: done

**Pull request merged**